### PR TITLE
feat: updating ami to debian-11 and renaming instance name to include env

### PR DESCRIPTION
**Pull request by marcus-girolneto**

### CODE REVIEW

File: ec2-cluster/modules/node/main.tf [hunk 1]
```diff
@@ -33,7 +33,7 @@ resource "aws_instance" "node" {
 
   tags = {
     Index     = count.index
-    Name      = "node-${var.cluster_id}-${count.index}"
+    Name      = "node-${var.env}-${var.cluster_id}-${count.index}"
```
Comment (marcinwadon) [inline ec2-cluster/modules/node/main.tf:36]: I think the shorter the better, having all tags available to display.
image

File: ec2-cluster/modules/cluster-provisioner/main.tf [hunk 2]
```diff
@@ -26,6 +26,7 @@ resource "null_resource" "cluster_provisioner" {
 
   provisioner "remote-exec" {
     inline = [
+      "sudo apt-get install jq -y",
```
Comment (marcinwadon) [inline ec2-cluster/modules/cluster-provisioner/main.tf:29]: If you modify it, please add `htop` and `lnav`, as both are heavily used too

**Review by marcinwadon — CHANGES_REQUESTED** [review]

**Comment by marcinwadon** [comment]
> from our discussion yesterday, is it possible to install `sdkman` and install Java 11 through that tool instead of through the Debian package manager?

why it should be a preferred way?

**Comment by scasplte2** [comment]
> > from our discussion yesterday, is it possible to install `sdkman` and install Java 11 through that tool instead of through the Debian package manager?
> 
> why it should be a preferred way?



> > from our discussion yesterday, is it possible to install `sdkman` and install Java 11 through that tool instead of through the Debian package manager?
> 
> why it should be a preferred way?

Marcus said he was forced to use a previous version of Debian as the latest version removes java-11 from the default package manager (which I confirmed from this post https://www.reddit.com/r/debian/comments/14fjgus/need_help_installing_openjdk11_on_debian_12/). I was suggesting to use `sdkman` to install Java 11 instead since I was under the impression that these changes were motivated at getting the images up-to-date in the first place.

File: ec2-cluster/modules/cluster-provisioner/main.tf (hunk 2 above)
Comment (IPadawans) [reply ec2-cluster/modules/cluster-provisioner/main.tf:29]: 👍🏼

File: ec2-cluster/modules/node/main.tf (hunk 1 above)
Comment (IPadawans) [reply ec2-cluster/modules/node/main.tf]: 👍🏼

**Review by marcinwadon — APPROVED** [review]
LGTM but install command can be simplified to one line

**Review by kpudlik — APPROVED** [review]

**Pull request merged**